### PR TITLE
#170947493 User can reset his password

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
- add reset password feature
#### Description of Task to be completed?
- added component for the feature
- added actions and reducers for the feature
- added storybook stories for the feature
- added tests for the feature

#### How should this be manually tested?
- clone repo
- run `npm run start:dev`
- use the following URL: `http://localhost:3000/password/forgot`
- provide valid email to which instructions to reset the pasword will be sent

#### What are the relevant pivotal tracker stories?
- [#170947493](https://www.pivotaltracker.com/story/show/170947493)
#### Screenshots (if appropriate)
forgot
reset

#### Questions:

### CODE REVIEW

**Review by izzett222 — CHANGES_REQUESTED** [review]
Screen Shot 2020-05-18 at 11 00 23
the reset page is not being found

**Comment by Baraka-Mugisha** [comment]
> Screen Shot 2020-05-18 at 11 00 23
> 
> the reset page is not being found

That is because the frontend link on the backend takes us to the main `develop app` and not the `review app`. For now, we can just replace `https://devrepublic-bn.herokuapp.com/` with `https://devrepublic-ft-password-yy7b7j.herokuapp.com/`

**Review by Ntare22 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

File: src/__tests__/forgotPassword.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable no-console */
```
Comment (izzett222) [inline src/__tests__/forgotPassword.test.js]: could you remove this line which disable the eslint rule

File: src/__tests__/loginRedux.test.js [hunk 2]
```diff
@@ -1,6 +1,4 @@
 /* eslint-disable no-console */
```
Comment (izzett222) [inline src/__tests__/loginRedux.test.js]: could you remove this line as you are not using console in the file

File: src/__tests__/resetPassword.test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable no-console */
```
Comment (izzett222) [inline src/__tests__/resetPassword.test.js]: could you remove this line which disable the eslint rule

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/__tests__/loginRedux.test.js (hunk 2 above)
Comment (Baraka-Mugisha) [reply src/__tests__/loginRedux.test.js]: I have removed it

**Review by izzett222 — APPROVED** [review]

File: src/__tests__/forgotPassword.test.js (hunk 1 above)
Comment (Baraka-Mugisha) [reply src/__tests__/forgotPassword.test.js]: I have removed the console

File: src/__tests__/resetPassword.test.js (hunk 3 above)
Comment (Baraka-Mugisha) [reply src/__tests__/resetPassword.test.js]: I have removed it

**Comment by jabichris** [comment]
Hi @Baraka-Mugisha , I am able to get the email asking me to change the password but when I click on the link, This below is what I am getting. Can you make sure the redirect link is redirecting me where I should change the password?
Screen Shot 2020-05-20 at 14 54 42

**Comment by Baraka-Mugisha** [comment]
> Hi @Baraka-Mugisha , I am able to get the email asking me to change the password but when I click on the link, This below is what I am getting. Can you make sure the redirect link is redirecting me where I should change the password?
> Screen Shot 2020-05-20 at 14 54 42

I have fixed the issue.

**Pull request merged**